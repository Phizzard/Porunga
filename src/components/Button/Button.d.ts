import { Rounded, Palette } from '../../types'
/**
 * Button properties
 */
export interface ButtonProps {
  /** Content inside the button element */
  onClick?(): any
  disabled?: boolean
  className?: string
  primaryColor?: Palette
  secondaryColor?: Palette
  isOutline?: boolean
  rounded?: Rounded
  label: string
}
/**
 * Button
 */
export declare const Button: ({
  onClick,
  disabled,
  className,
  primaryColor,
  secondaryColor,
  isOutline,
  label,
  rounded,
}: ButtonProps) => JSX.Element
export default Button
