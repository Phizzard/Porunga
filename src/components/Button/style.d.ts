/// <reference types="react" />
import { SerializedStyles } from '@emotion/core'
interface ButtonProps {
  primaryColor?: string
  secondaryColor?: string
  rounded?: SerializedStyles
  isOutline?: boolean
}
export declare const Container: import('@emotion/styled-base').StyledComponent<
  import('react').DetailedHTMLProps<
    import('react').ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  >,
  ButtonProps,
  object
>
export {}
