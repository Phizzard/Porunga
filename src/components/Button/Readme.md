Default Button
```jsx
<Button>Default Button</Button>
```