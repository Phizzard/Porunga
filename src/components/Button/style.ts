import styled from '@emotion/styled'

interface ButtonProps {
  backgroundColor?: string
}

export const Container = styled.button<ButtonProps>`
  color: ${p => p.backgroundColor};
  padding: 0.6rem 1.1rem;
  font-size: 1.1rem;
  font-weight: 500;
  border: 3px solid ${p => p.backgroundColor};
  border-radius: 24px;
  cursor: pointer;
  background-color: #f7fafc;
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06);
  transition: 0.2s;
  :hover {
    background-color: ${p => p.backgroundColor};
    color: #fff;
  }
`
