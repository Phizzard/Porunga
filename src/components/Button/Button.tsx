import React from "react";
import { Container } from "./style";

/**
 * Button properties
 */
export interface IButtonProps {
  /** children description */
  children?: React.ReactNode;
  /** className description */
  className: string;
};

/**
 * Button
 */
export const Button = ({ children, className }: IButtonProps) => (
  <Container className={className}>{children}</Container>
)

export default Button;