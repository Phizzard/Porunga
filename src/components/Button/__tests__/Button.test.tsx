import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../Button'

it('renders correctly', () => {
  const component = renderer.create(<Button />)

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
