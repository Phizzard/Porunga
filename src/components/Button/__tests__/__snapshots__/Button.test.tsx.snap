// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  aria-label="test"
  className="css-1mjt4j"
>
  test
</button>
`;
